src/frontend/src/optimizer/rule/apply_offset_rewriter.rs

```diff
@@ -70,3 +68,19 @@ impl ApplyOffsetRewriter {
         self.has_correlated_input_ref = false;
     }
 }
+
+pub struct ApplyCorrelatedIndicesConverter {}
+
+impl ApplyCorrelatedIndicesConverter {
+    pub fn convert_to_index_mapping(correlated_indices: &[usize]) -> ColIndexMapping {
```

Comment:
Currently, `correlated_indices` is not definitely invertible, so we need that logic.